### PR TITLE
Added "absent" matching strategy for query parameters

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy [hunk 1]
```diff
@@ -695,6 +695,8 @@ class WiremockGroovyDslSpec extends WiremockSpec {
 							parameter 'search': $(client(notMatching(~/^\/[0-9]{2}$/)), server("10"))
 							parameter 'age': $(client(notMatching("^\\w*\$")), server(10))
 							parameter 'name': $(client(matching("Denis.*")), server("Denis"))
+							parameter 'credit': $(client(absent()), server("123"))
```
Comment (marcingrzejszczak) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy]: if it's absent then why is there a value for the other side? I don't follow :/

Comment (dstepanov) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy]: Yes, I have just added line for testing without thinking :D it only makes sense absent for all sides

Comment (marcingrzejszczak) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy]: You should ensure that the test fails if you do so - current implementation is not ok :/

**Comment by dstepanov** [comment]
Added validation

File: accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy [hunk 2]
```diff
@@ -863,6 +867,62 @@ class WiremockGroovyDslSpec extends WiremockSpec {
             e.message.contains "Query parameter 'age' can't be of a matching type: NOT_MATCHING for the server side"
 	}
 
+	def "should not allow query parameter with a different absent variation for server/client"() {
```
Comment (marcingrzejszczak) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy:870]: This you could have written with the `where:` block where you would have the result of the `GroovyDsl.make` method. WDYT?

Comment (dstepanov) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy:870]: Nope, there is no result of `GroovyDsl.make` - an exception is throws, right?

Comment (dstepanov) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy:870]: I can try to use a closure :P

Comment (dstepanov) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy:870]: @marcingrzejszczak  What would you say if it looked like:

``` groovy
def "should not allow query parameter with a different absent variation for server/client"() {
        when:
            GroovyDsl.make dsl
        then:
            def e = thrown(IllegalStateException)
            e.message.contains "Absent cannot only be used only on one side"
        where:
            dsl << [
               {
                    request {
                        method 'GET'
                        urlPath("users") {
                            queryParameters {
                                parameter 'name': $(client(absent()), server(""))
                            }
                        }
                    }
                    response {
                        status 200
                    }
                },
                {
                    request {
                        method 'GET'
                        urlPath("users") {
                            queryParameters {
                                parameter 'name': $(client(""), server(absent()))
                            }
                        }
                    }
                    response {
                        status 200
                    }
                },
                {
                    request {
                        method 'GET'
                        urlPath("users") {
                            queryParameters {
                                parameter 'name': $(client(absent()), server(matching("abc")))
                            }
                        }
                    }
                    response {
                        status 200
                    }
                }
            ]
    }
```

Editor highlighting is broken, the second option is to use lazy evaluation using closure:

``` groovy
def "should not allow query parameter with a different absent variation for server/client"() {
        when:
            make()
        then:
            def e = thrown(IllegalStateException)
            e.message.contains "Absent cannot only be used only on one side"
        where:
            make << [

                { ->
                    GroovyDsl.make {
                        request {
                            method 'GET'
                            urlPath("users") {
                                queryParameters {
                                    parameter 'name': $(client(absent()), server(""))
                                }
                            }
                        }
                        response {
                            status 200
                        }
                    }
                },

                { ->
                    GroovyDsl.make {
                        request {
                            method 'GET'
                            urlPath("users") {
                                queryParameters {
                                    parameter 'name': $(client(""), server(absent()))
                                }
                            }
                        }
                        response {
                            status 200
                        }
                    }
                },

                { ->
                    GroovyDsl.make {
                        request {
                            method 'GET'
                            urlPath("users") {
                                queryParameters {
                                    parameter 'name': $(client(absent()), server(matching("abc")))
                                }
                            }
                        }
                        response {
                            status 200
                        }
                    }
                }
            ]
    }
```

Comment (marcingrzejszczak) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy:870]: I like the first option - WDYT?

Comment (dstepanov) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy:870]: OK I'll change it, but it looks ugly in IDEA ;)

Comment (dstepanov) [inline accurest-core/src/test/groovy/io/codearte/accurest/dsl/WiremockGroovyDslSpec.groovy:870]: @marcingrzejszczak Done

**Pull request merged**